Big merge

albnunes opened the pull request.

Commit:
feat: changed static function updateStatus to public to use it in timeForTheTruth.ts

Commit:
feat: made error treatment better

Commit:
feat: implemented timeForTheTruth.html and   timeForTheTruth.ts

Commit:
feat: updated imports from Meeta and finished logic to change challenge status

Commit:
feat: created pug file for timeForTheTruth

Commit:
feat: crated css file to change the margin and finished html of the task

rairezende committed:
rai turned the html and css into pugs

feat: updated pug and css

feat: updated pug and css

chore(deadline monitor and challenge status): fix things in the deadlinemonitor to work

refactor(challenge created web page): cleaning the code and the style of the page

feat(husky,package.json.): check if the code in the staged area is within the project's standard

Check if the code in the staged area is within the project's standard.

Update challenge.entity.ts

chore: missing bracket

Update challenge.entity.ts

method updateStatus has been deleted. I added it again

refactor: refactor

 ts files e pug files para envio dos e-mails

Co-authored-by: Raquel <raquel@toki.life>

.pug youWereChallenged

img folder

created imgs folder

temporary teste folder for templates

temporary teste folder in templates

perf(emails templates): did a base template for all emails

import pug in ts file and compile

sync commit

html merged

mixin base template

base-style

naming improve

deadlineEmail template update

no more necessary folders deleted

email pics updated

added 3 email templates

templates index ts commented

templates updated n width to 740

hoodie gif deleted

chore(template emails): made templates for the emails we send

created btn mixin and added to templates

added btn mixin to templates

refactor(template emails): cleaning the code and exporting the classes from email

pics renamed

feat: did email classes that were missing

added btn-container and updated pug files

css and pug files improved

repo maint. cmmd updated and index n supCon upd

player and challenge repo entity created

IChallengeRepository's import path updated

fix(lib/domain): removes js files on domain

Probably an artefact of a stray TSC

feature: interface and symbols for repository finished

defined Dependency Injection in app module

inject the symbols in repositories

implemented interfaces in repositories and updated import's

updates in challenge and player indexes

fix: fix the imports in interfaces

Alberto/atqr 94: Adds Updates Challenge Endpoint (#32)

* refactor: entity method that updates status

* feature: I used the notification service as a base to notify the player

* refactor: redid the endpoint more correct than before

* fix(api): fixes minor mistakes created after rebasing

Co-authored-by: Yago Fernandes <yagoalmeida@gmail.com>

chore(domain): adds index files to domain

Alberto/atqr 92: Refactor Update challenge methods (#31)

* feature: create a global dto to endpoint patch

* implement the methods in endpoint

* the methods in front are updated

* refactor: Passed the logic to a middleware

* formatted document

* feat(controller): added a route for each method in controller

added a different patch route for method in controller

* refactor(atqr api): adds api indexes and absolutes paths to api and Tsconfig

* refactor: i removed useless imports and cleaned the code

* fix(api/dto): renames Update Challenge Dto

* chore(api): fixes spelling mistakes and other minor fixes

Co-authored-by: Yago Fernandes <yagoalmeida@gmail.com>

refactor(terms-style.scss): changed to css to scss

build(yarn run for catch tools): runed yarn

feat: changed kebab cases for camelCases and ajusted mixin terms's indentation

refactor(web): pages are now running and without compiled files

Commit:
pull request changes: deleted html files, imported cdns and ts files on timeFortheTruth.pug

Commit:
finished reviewing task

Commit:
resolve: did changes that ygo pointed on PR

rairezende committed:
Email template style from pug to css (Atqr-111) (#44)

* Alberto/atqr 92: Refactor Update challenge methods (#31)

* feature: create a global dto to endpoint patch

* implement the methods in endpoint

* the methods in front are updated

* refactor: Passed the logic to a middleware

* formatted document

* feat(controller): added a route for each method in controller

added a different patch route for method in controller

* refactor(atqr api): adds api indexes and absolutes paths to api and Tsconfig

* refactor: i removed useless imports and cleaned the code

* fix(api/dto): renames Update Challenge Dto

* chore(api): fixes spelling mistakes and other minor fixes

Co-authored-by: Yago Fernandes <yagoalmeida@gmail.com>

* feat: added pug style file that has style css

* feat: uses camelCase (all files names and content) and relative units (in scss)

* fix(domain): path to template is fixed

Co-authored-by: Alberto Nunes <bisneto.alberto@gmail.com>
Co-authored-by: Yago Fernandes <yagoalmeida@gmail.com>

albnunes committed:
fix: import's in domain are resolved

GabrielaRsanches committed:
Gabi/atqr 7 page hora da verdade (#42)

* feat: changed static function updateStatus to public to use it in timeForTheTruth.ts

* feat: made error treatment better

* feat: implemented timeForTheTruth.html and   timeForTheTruth.ts

* feat: updated imports from Meeta and finished logic to change challenge status

* feat: created pug file for timeForTheTruth

* pull request changes: deleted html files, imported cdns and ts files on timeFortheTruth.pug

* finished reviewing task

* resolve: did changes that ygo pointed on PR

* refactor(apiclient.ts): file name is now apiClient, not api

* refactor(timeforthetruth.ts and js): congrats and reprove buttons use clientApi to change status

uses getUrlId() from urlDetails

BREAKING CHANGE: web is now quering strings from browser urls

* refactor(urldetails.ts and js on web): creation of urlDetails to implement getUrlId()

gets url from browser and extraxts the id of it. It is on a separate file to be reused and the file
to hold funcions linked to url

* refactor(wip): wIP

* refactor(deadline.ts, deadline.pug and deadlineindex): timeForTheTruth is called dealine. Page is ok

timeForTheTruth is now called dealine. File ts is now using generic function updateUi from services
to display the information. Pug is ready and appling cdns imports

* refactor(updateui.ts): generic function to get elements with querySelectorAll and display

* refactor(urldetails.ts): function is now generic

* refactor(apiclient.ts and interfaces.ts): interfaces.ts holds interfaces to be used in frontend

ApiClient now uses interfaces that represent the actual classes that have to implement them in the
future

BREAKING CHANGE: frontend now uses interfaces instead of the actual classes

* refactor(challenge, challengeindex, createchallenge, lastplayerssession and index (all ".pug")): pag

pug pages are compiling and running. Challenge.pug renders challenge resume, challengeIndex has its
content.

* refactor(files changed that have nothing to do with the task): files changed

dont know why some changes happend but im commiting because I fear making a mass otherwise

* style(js files): js file were deleted

* refactor(web): pug files are running

* refactor(web): path to components is fixed

Co-authored-by: Alberto Nunes <bisneto.alberto@gmail.com>

albnunes committed:
style: style fix

albnunes committed:
Alberto/atqr 87 endpoint update supervisor (#45)

* Alberto/atqr 92: Refactor Update challenge methods (#31)

* feature: create a global dto to endpoint patch

* implement the methods in endpoint

* the methods in front are updated

* refactor: Passed the logic to a middleware

* formatted document

* feat(controller): added a route for each method in controller

added a different patch route for method in controller

* refactor(atqr api): adds api indexes and absolutes paths to api and Tsconfig

* refactor: i removed useless imports and cleaned the code

* fix(api/dto): renames Update Challenge Dto

* chore(api): fixes spelling mistakes and other minor fixes

Co-authored-by: Yago Fernandes <yagoalmeida@gmail.com>

* refactor: added export's in classes

* feat: added export's in classes

* feat: improved supervisor enum and created method to update the supervisor status

I created a method to update the supervisor status and I improved the supervisor enum

* feat: created endpoint update supervisor

* refactor(review pr): comments explaning code

Co-authored-by: Yago Fernandes <yagoalmeida@gmail.com>
Co-authored-by: Gabriela Sanches <gabi@toki.life>

albnunes committed:
Alberto/atqr-93-elaborate-api-errors-refactor (#33)

* rai turned the html and css into pugs

feat: updated pug and css

feat: updated pug and css

chore(deadline monitor and challenge status): fix things in the deadlinemonitor to work

refactor(challenge created web page): cleaning the code and the style of the page

feat(husky,package.json.): check if the code in the staged area is within the project's standard

Check if the code in the staged area is within the project's standard.

Update challenge.entity.ts

chore: missing bracket

Update challenge.entity.ts

method updateStatus has been deleted. I added it again

refactor: refactor

 ts files e pug files para envio dos e-mails

Co-authored-by: Raquel <raquel@toki.life>

.pug youWereChallenged

img folder

created imgs folder

temporary teste folder for templates

temporary teste folder in templates

perf(emails templates): did a base template for all emails

import pug in ts file and compile

sync commit

html merged

mixin base template

base-style

naming improve

deadlineEmail template update

no more necessary folders deleted

email pics updated

added 3 email templates

templates index ts commented

templates updated n width to 740

hoodie gif deleted

chore(template emails): made templates for the emails we send

created btn mixin and added to templates

added btn mixin to templates

refactor(template emails): cleaning the code and exporting the classes from email

pics renamed

feat: did email classes that were missing

added btn-container and updated pug files

css and pug files improved

repo maint. cmmd updated and index n supCon upd

player and challenge repo entity created

IChallengeRepository's import path updated

fix(lib/domain): removes js files on domain

Probably an artefact of a stray TSC

feature: interface and symbols for repository finished

defined Dependency Injection in app module

inject the symbols in repositories

implemented interfaces in repositories and updated import's

updates in challenge and player indexes

fix: fix the imports in interfaces

Alberto/atqr 94: Adds Updates Challenge Endpoint (#32)

* refactor: entity method that updates status

* feature: I used the notification service as a base to notify the player

* refactor: redid the endpoint more correct than before

* fix(api): fixes minor mistakes created after rebasing

Co-authored-by: Yago Fernandes <yagoalmeida@gmail.com>

chore(domain): adds index files to domain

Alberto/atqr 92: Refactor Update challenge methods (#31)

* feature: create a global dto to endpoint patch

* implement the methods in endpoint

* the methods in front are updated

* refactor: Passed the logic to a middleware

* formatted document

* feat(controller): added a route for each method in controller

added a different patch route for method in controller

* refactor(atqr api): adds api indexes and absolutes paths to api and Tsconfig

* refactor: i removed useless imports and cleaned the code

* fix(api/dto): renames Update Challenge Dto

* chore(api): fixes spelling mistakes and other minor fixes

Co-authored-by: Yago Fernandes <yagoalmeida@gmail.com>

refactor(terms-style.scss): changed to css to scss

build(yarn run for catch tools): runed yarn

feat: changed kebab cases for camelCases and ajusted mixin terms's indentation

refactor(web): pages are now running and without compiled files

* rai turned the html and css into pugs

feat: updated pug and css

feat: updated pug and css

chore(deadline monitor and challenge status): fix things in the deadlinemonitor to work

refactor(challenge created web page): cleaning the code and the style of the page

feat(husky,package.json.): check if the code in the staged area is within the project's standard

Check if the code in the staged area is within the project's standard.

Update challenge.entity.ts

chore: missing bracket

Update challenge.entity.ts

method updateStatus has been deleted. I added it again

refactor: refactor

 ts files e pug files para envio dos e-mails

Co-authored-by: Raquel <raquel@toki.life>

.pug youWereChallenged

img folder

created imgs folder

temporary teste folder for templates

temporary teste folder in templates

perf(emails templates): did a base template for all emails

import pug in ts file and compile

sync commit

html merged

mixin base template

base-style

naming improve

deadlineEmail template update

no more necessary folders deleted

email pics updated

added 3 email templates

templates index ts commented

templates updated n width to 740

hoodie gif deleted

chore(template emails): made templates for the emails we send

created btn mixin and added to templates

added btn mixin to templates

refactor(template emails): cleaning the code and exporting the classes from email

pics renamed

feat: did email classes that were missing

added btn-container and updated pug files

css and pug files improved

repo maint. cmmd updated and index n supCon upd

player and challenge repo entity created

IChallengeRepository's import path updated

fix(lib/domain): removes js files on domain

Probably an artefact of a stray TSC

feature: interface and symbols for repository finished

defined Dependency Injection in app module

inject the symbols in repositories

implemented interfaces in repositories and updated import's

updates in challenge and player indexes

fix: fix the imports in interfaces

Alberto/atqr 94: Adds Updates Challenge Endpoint (#32)

* refactor: entity method that updates status

* feature: I used the notification service as a base to notify the player

* refactor: redid the endpoint more correct than before

* fix(api): fixes minor mistakes created after rebasing

Co-authored-by: Yago Fernandes <yagoalmeida@gmail.com>

chore(domain): adds index files to domain

Alberto/atqr 92: Refactor Update challenge methods (#31)

* feature: create a global dto to endpoint patch

* implement the methods in endpoint

* the methods in front are updated

* refactor: Passed the logic to a middleware

* formatted document

* feat(controller): added a route for each method in controller

added a different patch route for method in controller

* refactor(atqr api): adds api indexes and absolutes paths to api and Tsconfig

* refactor: i removed useless imports and cleaned the code

* fix(api/dto): renames Update Challenge Dto

* chore(api): fixes spelling mistakes and other minor fixes

Co-authored-by: Yago Fernandes <yagoalmeida@gmail.com>

refactor(terms-style.scss): changed to css to scss

build(yarn run for catch tools): runed yarn

feat: changed kebab cases for camelCases and ajusted mixin terms's indentation

refactor(web): pages are now running and without compiled files

* Email template style from pug to css (Atqr-111) (#44)

* Alberto/atqr 92: Refactor Update challenge methods (#31)

* feature: create a global dto to endpoint patch

* implement the methods in endpoint

* the methods in front are updated

* refactor: Passed the logic to a middleware

* formatted document

* feat(controller): added a route for each method in controller

added a different patch route for method in controller

* refactor(atqr api): adds api indexes and absolutes paths to api and Tsconfig

* refactor: i removed useless imports and cleaned the code

* fix(api/dto): renames Update Challenge Dto

* chore(api): fixes spelling mistakes and other minor fixes

Co-authored-by: Yago Fernandes <yagoalmeida@gmail.com>

* feat: added pug style file that has style css

* feat: uses camelCase (all files names and content) and relative units (in scss)

* fix(domain): path to template is fixed

Co-authored-by: Alberto Nunes <bisneto.alberto@gmail.com>
Co-authored-by: Yago Fernandes <yagoalmeida@gmail.com>

* Gabi/atqr 7 page hora da verdade (#42)

* feat: changed static function updateStatus to public to use it in timeForTheTruth.ts

* feat: made error treatment better

* feat: implemented timeForTheTruth.html and   timeForTheTruth.ts

* feat: updated imports from Meeta and finished logic to change challenge status

* feat: created pug file for timeForTheTruth

* pull request changes: deleted html files, imported cdns and ts files on timeFortheTruth.pug

* finished reviewing task

* resolve: did changes that ygo pointed on PR

* refactor(apiclient.ts): file name is now apiClient, not api

* refactor(timeforthetruth.ts and js): congrats and reprove buttons use clientApi to change status

uses getUrlId() from urlDetails

BREAKING CHANGE: web is now quering strings from browser urls

* refactor(urldetails.ts and js on web): creation of urlDetails to implement getUrlId()

gets url from browser and extraxts the id of it. It is on a separate file to be reused and the file
to hold funcions linked to url

* refactor(wip): wIP

* refactor(deadline.ts, deadline.pug and deadlineindex): timeForTheTruth is called dealine. Page is ok

timeForTheTruth is now called dealine. File ts is now using generic function updateUi from services
to display the information. Pug is ready and appling cdns imports

* refactor(updateui.ts): generic function to get elements with querySelectorAll and display

* refactor(urldetails.ts): function is now generic

* refactor(apiclient.ts and interfaces.ts): interfaces.ts holds interfaces to be used in frontend

ApiClient now uses interfaces that represent the actual classes that have to implement them in the
future

BREAKING CHANGE: frontend now uses interfaces instead of the actual classes

* refactor(challenge, challengeindex, createchallenge, lastplayerssession and index (all ".pug")): pag

pug pages are compiling and running. Challenge.pug renders challenge resume, challengeIndex has its
content.

* refactor(files changed that have nothing to do with the task): files changed

dont know why some changes happend but im commiting because I fear making a mass otherwise

* style(js files): js file were deleted

* refactor(web): pug files are running

* refactor(web): path to components is fixed

Co-authored-by: Alberto Nunes <bisneto.alberto@gmail.com>

* style: style fix

* Alberto/atqr 92: Refactor Update challenge methods (#31)

* feature: create a global dto to endpoint patch

* implement the methods in endpoint

* the methods in front are updated

* refactor: Passed the logic to a middleware

* formatted document

* feat(controller): added a route for each method in controller

added a different patch route for method in controller

* refactor(atqr api): adds api indexes and absolutes paths to api and Tsconfig

* refactor: i removed useless imports and cleaned the code

* fix(api/dto): renames Update Challenge Dto

* chore(api): fixes spelling mistakes and other minor fixes

Co-authored-by: Yago Fernandes <yagoalmeida@gmail.com>

* resolve: did changes that ygo pointed on PR

* feature: global exception filter to send http errors

* use decorator to can use filters in controlller

* refactor: remade the filter to take only treated error

* refactor(hhtpexceptions and challenge.controller): all imports are been used

Co-authored-by: Raiane <raiane.rze@gmail.com>
Co-authored-by: Raiane Rezende <90356301+rairezende@users.noreply.github.com>
Co-authored-by: Yago Fernandes <yagoalmeida@gmail.com>
Co-authored-by: Gabriela Sanches <86623626+GabrielaRsanches@users.noreply.github.com>
Co-authored-by: Gabriela Sanches <gabi@toki.life>

angeliralves committed:
feat(web/components): i added the information where the money goes (#46)

albnunes committed:
refactor: unused files were deleted and the others were cleaned

GabrielaRsanches committed:
Gabi/atqr 117 refactor repository (#50)

* ci(yarn file): changes were commited

* refactor(challenge and player repositories and its interfaces): functions are all asynchronous

GabrielaRsanches committed:
refactor notificationservice.ts  (#47)

* refactor(email.interface.ts, email.service.ts): propriety "to" of both classes has new types

propriety "to" expects now also Player['_email'] | Challenge['_supervisorEmail'] because of emails
that has to be sent to supervisor

* refactor(challengestarted, congrats, deadlineemail and paytheprice.ts): classes have different types

classes of email templates have now on propety "to" Player['_email'] or Challenge["supervisorEmail"]
been expected

* refactor(confirmemail.ts): player import is now working

* refactor(notification.service.ts): added comments and changed notifyCompletedChallenges()

added comments to describe what is happening in code. Changed notifyOverdueChallenges() email to
DeadLineEmail after change in class DeadLineEmail  to be sent to supervisor intead of player

* refactor(types used in email service): email and notification service and email props in Entitys

provided right types

* refactor(api and domain): challenge controller , dto and repository have correct email types

* fix: notification service is running

Co-authored-by: Alberto Nunes <bisneto.alberto@gmail.com>

yfernandes commented on apps/atqr/api/src/app/filter/httpExceptions.ts:
```diff
@@ -0,0 +1,45 @@
+    }
+    if (exception.message.length <= 10) {
+      console.log('Your message is too short. Do a better one');
+      exception = new InternalServerErrorException(
```
The console log and error messages are contradictory. We know what happened, the message was to short, throw the appropriate error

yfernandes commented on apps/atqr/api/src/app/infra/email/mailer.service.ts:
```diff
@@ -34,7 +34,7 @@ export class Mailer implements OnApplicationBootstrap, IMailer {
     const messageData = {
       from: this.fromEmail,
-      to: email.playerEmail,
+      to: email,
```
Incorrect Type, email refers to a Email Object,  and the to property expects a string with a Email Address

yfernandes commented on apps/atqr/api/src/app/challenge.controller.ts:
```diff
@@ -87,8 +94,8 @@ export class ChallengeController {
         // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        const emailAddress = new EmailAddress(challengeDto.supervisorEmail);
-        const email = new ChallengeStarted(player);
+        const emailAddress = challengeDto.supervisorEmail;
```
Unused variable should be deleted

yfernandes commented on apps/atqr/api/src/app/challenge.controller.ts:
```diff
@@ -120,16 +127,62 @@ export class ChallengeController {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  changeSupervisor(@Param('id') id: Guid): void {
-    return;
+  //"update"s the result information of a request in the specified route
```
Remove unnecessary comments, semantic code speaks for itself

yfernandes commented on apps/atqr/web/src/components/challengeSuggestions/suggestionFunction.ts:
```diff
@@ -1,3 +1,4 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
There are multiple errors, and bad practices within this file. Further comments will be provided lather

albnunes replied:
I refactored this code, but I don't know what the bad practices would be. point out the errors please

yfernandes commented on apps/atqr/web/src/components/createChallenge/createChallenge.pug:
```diff
@@ -1,151 +1,120 @@
+                                .col-md-12
+                                  // Single Form Start
+                                  .single-form
+                                    textarea#descricao.form-control(placeholder='Descreva Seu Desafio' name ='challengeYourselfDescription')
```
All classes and ids must be written in english.

yfernandes commented on apps/atqr/web/src/components/lastPlayersSession/getLastPlayers.ts:
```diff
@@ -1,13 +1,15 @@
-  }
-}
-getLastChallenges();
+//TODO: this function should be using apiClient instead of axios directly
```
Commented code should either be deleted or implement the task in the todo

yfernandes commented on apps/atqr/web/src/components/lastPlayersSession/lastPlayersSession.pug:
```diff
@@ -36,4 +36,4 @@
                 span.swiper-pagination-bullet.swiper-pagination-bullet-active(tabindex='0' role='button' aria-label='Go to slide 3')
               span.swiper-notification(aria-live='assertive' aria-atomic='true')
-script(type="module" src="../../components/lastPlayersSession/getLastPlayers.ts")
+//script(type="module" src="./getLastPlayers.ts")
```
Commented code should be deleted or fixed

yfernandes commented on apps/atqr/web/src/components/createChallenge/createChallenge.pug:
```diff
@@ -1,151 +1,120 @@
+          .meeta-event-schedule-tabs
+            // Event Schedule Tabs Menu Start
+            ul.meeta-event-schedule-tabs-menu
+              li.active(data-tab-target='#tab1' id='yourself')
```
Ids must have meaninfull names

yfernandes commented on apps/atqr/web/src/components/createChallenge/createChallenge.pug:
```diff
@@ -1,151 +1,120 @@
+              li.active(data-tab-target='#tab1' id='yourself')
+                .meeta-event-schedule-time
+                  h4.meeta-event-schedule-day Se Desafie !
+              li.tab(data-tab-target='#tab2' id='someone')
```
The id someone has no semantic value, it has no context from where it's used, how its used. Needs refactoring

yfernandes replied:
Same for "Yourself above"

yfernandes commented on apps/atqr/web/src/pages/challenge/challenge.ts:
```diff
@@ -1,47 +1,22 @@
-  supervisorEmail,
+export const getChallenge = () => {
+  const data = atqrApi.challenges.getOne('');
+  console.log('this is data :', data);
```
When loggin data be carefull of the message you craft. "this is data" without context in the terminal means nothing.

yfernandes commented on apps/atqr/web/src/pages/challenge/challengeMixins.pug:
```diff
@@ -0,0 +1,15 @@
+mixin headScripts
```
Is a one line mixin necessary?

yfernandes commented on apps/atqr/web/src/pages/deadline/deadlineMixins.pug:
```diff
@@ -0,0 +1,12 @@
+mixin headScripts
```
Again, is a one line mixin really necessary?

yfernandes commented on apps/atqr/web/src/services/apiClient.ts:
```diff
@@ -35,7 +34,7 @@ export const atqrApi = {
       }
     },
     challengeUpdated: {
```
Challenge properties names needs to be reworked to be semantic. While inside challenges its own property doesn't have to be called challenge updated, it can be called only by update. The same logic follows for the entirety of the file

yfernandes commented on apps/atqr/web/src/services/urlDetails.ts:
```diff
@@ -0,0 +1,10 @@
+const url = window.location.pathname;
+const urlSearchParams = new URLSearchParams(url);
+export async function getUrlId(props) {
```
Untyped parameter, and unnecessary await. Both those error can and are statically caught by vscode.
The parameter of all functions created by us must be STRONGLY typed, and the use of await, and therefore async functions, is never by luck.
You can always know if Async and Await are needed

yfernandes commented on package.json:
```diff
@@ -88,13 +90,15 @@
     "@types/pug": "^2.0.6",
     "@typescript-eslint/eslint-plugin": "5.3.0",
     "@typescript-eslint/parser": "5.3.0",
+    "assert": "^2.0.0",
```
Wildcard selector for npm packages is HIGHLY discouraged

yfernandes commented on libs/atqr/domain/src/lib/challenge/index.ts:
```diff
@@ -1,4 +1,3 @@
-
+export * from './challenge.entity';
+export * from './challengeRepo.interface';
+export * from '../services';
```
A local index file should not export a file outside itself

yfernandes commented on libs/atqr/domain/src/lib/player/index.ts:
```diff
@@ -1,2 +1,4 @@
+//export * from './player.entity';
+export * from './playerRepo.interface';
```
Avoid abreviations. Change spelling to playerRepository

yfernandes commented on libs/atqr/domain/src/lib/EmailService/email.interface.ts:
```diff
@@ -1,7 +1,8 @@
 
 export interface IEmail {
-  to: Player;
+  to: Player | Player['_email'] | Challenge['_supervisorEmail'];
```
Looks like this compound type is telling anothes history. It's telling me that it wants to be of type EmailAddress

yfernandes commented on libs/atqr/domain/src/lib/EmailService/templates/youWereChallenged.pug:
```diff
@@ -1,10 +1,11 @@
   body
-    include includes/base-template
-    +email-content('Tã tã tã tã tã tã tã tã tã tã tã tã tã tã tã tã tã tãã tãããã 🎶', 'Interrompemos a programação normal da sua vida sem graça e sem sal para informar que  #{supervisor.name} te chamou pra briga. Vai encarar? VAI ENCARAR?',
+    include includes/base-template   
```
Update base template name to camelCase

yfernandes commented on libs/atqr/domain/src/lib/EmailService/templates/supDenied.ts:
```diff
@@ -2,14 +2,15 @@ import { Player } from '../../player';
 
-class SupervisorDenied extends Email {
-  constructor(to: Player) {
+export class SupervisorDenied extends Email {
```
The SupervisorDenied Class should not import the pugcompiler from the SupConfirmation file.
This class must be refactored

yfernandes commented on libs/atqr/domain/src/lib/EmailService/templates/challengeStarted.ts:
```diff
@@ -3,12 +3,16 @@ import { Email } from '../email.service';
 import * as pug from 'pug';
 import path = require('path');
 
+export const pathToTemplates = path.join(
```
A constant used throughout the project should be extracted to a separate file or to a configuration file.

yfernandes replied:
Is not the responsibility of the challenge started file to define and maintain the path to templates constant

yfernandes commented on libs/atqr/domain/src/lib/EmailService/templates/supAccepted.ts:
```diff
@@ -1,14 +1,13 @@
-import { Challenge } from "../../challenge";
 import { Player } from "../../player";
 import { Email } from "../email.service";
 import { pugFile } from "./supConfirmation";
```
A template compiler from another class CANNOT be imported. This files must be refactored

yfernandes commented on libs/atqr/domain/src/lib/EmailService/templates/congrats.ts:
```diff
@@ -2,14 +2,14 @@ import { Player } from '../../player/player.entity';
 export class Congrats extends Email {
-  constructor(to: Player) {
+  constructor(to: Player['_email']) {
     super(to,
```
Congrats email doesn't follow convention from other emails, template compiler must be passed to super in the constructor

yfernandes commented on libs/atqr/domain/src/lib/EmailService/templates/confirmEmail.ts:
```diff
@@ -1,13 +1,13 @@
 import { Email } from "../email.service";
 import { pugFile } from "./supConfirmation";
 
 class ConfirmEmail extends Email {
```
Class must be exported

yfernandes commented on libs/atqr/domain/src/lib/EmailService/templates/confirmEmail.ts:
```diff
@@ -1,13 +1,13 @@
-import { Player } from "../../player";
+import { Player } from "../../player/player.entity";
 import { Email } from "../email.service";
 import { pugFile } from "./supConfirmation";
```
A template compiler CANNOT be imported from another class!

yfernandes reviewed (commented):
This is only a shallow review, but it already has a few things to work on. Besides whats commented the entire project must be linted and formated. I'll let you work on that while I work on a more in depth review.

yfernandes commented on apps/atqr/web/src/components/challengeSuggestions/challengeSuggestions.pug:
```diff
@@ -26,9 +26,9 @@
                   span
                     a.register-btn.btn.btn-3.btn-outline-primary(onclick='challengeSuggestion(this)') Desafie alguém!
```
challengeSuggestion is undefined on the browser, this must be fixed.

albnunes committed:
refactor(web and back): web and back have been refactored

svallory commented:
@albnunes If you have fixed Yago's comments, please resolve the conversations and let me know when it's ready for review

albnunes committed:
refactor: resolved all comments except one

albnunes committed:
fix: some bugs that sonarcloud reported have been fixed

yfernandes committed:
Saulo/domain definition disl (#69)

* WIP

* WIP: Finishing DISL spec

* Add @disl/lang library with example code

* Add @disl/lang library with example code

* DISL: Work in Progress

* Co-authored-by: Saulo Vallory <saulo@toki.life>

* Raiane/adopt challenge disl (#53)

New additions to domain ATQR:

- Goals context
- Goal aggregate root
- visitorAdoptsGoal story

Co-authored-by: Saulo Vallory <saulo@toki.life>

* Add rules to verify if a Challenge can begin (#66)

* docs: method that every time it is called checks if the challenge can start
* docs: story to verify if the challenge can begin
* Final review

Co-authored-by: Alberto Nunes <albnunes@users.noreply.github.com>
Co-authored-by: Saulo Vallory <saulo@toki.life>

* Player receives email at the end of a Challenge (#61)

Sends an email to the player when the challenge finishes either congratulating or consoling them

Co-authored-by: Gabriela Sanches <gabriela@toki.life>
Co-authored-by: Raiane Rezende <raiane@toki.com>
Co-authored-by: Saulo Vallory <saulo@toki.life>

* fix(domain-spec): :bug: remove duplicated JudgeChallenge command

* Domain specification for ShowChallenge command (#64)

Implements the domain specification in disl for user story 0.3 [Show challenge](https://www.notion.so/tokinetwork/0-3-Ver-desafio-86dd983345934d9a993602ca2a0b2317)

Co-authored-by: Raquel Ellen <raquel@toki.life>
Co-authored-by: Saulo Vallory <saulo@toki.life>

* docs(domain-spec): Judge challenges a player story (#60)

See: https://www.notion.so/tokinetwork/1-2-Juiz-desafia-jogador-4aa8f0757d7d4001a516ad28d9c40a04

Co-authored-by: Gabriela Sanches <gabriela@toki.life>
Co-authored-by: Saulo Vallory <saulo@toki.life>

* docs: supervisor stories to domain-specs

* docs: documented the stories of the judge

* docs: refaactor

* docs: now the workflow is correct

* docs: requested changes have been made

* Update judgeAcceptsOfficiationRequest

* Update judgeRejectsOfficiationRequest.disl

* Update selectNewJudge

* docs(domain-spec): :necktie: add disl definition for the entire notification system

* docs(domain-spec): work on services and emails

- Split services.disl into multiple files
- Add business logic for resending transactional email

* docs(domain-spec): :necktie: select new judge

* docs(domain-spec): :necktie: reafctoring to split api.disl into multiple files

* docs(disl): flow of accepting and rejecting an officiation (#68)

* docs(disl): flow of accepting and rejecting an officiation

* chore: 🤖 final review

Co-authored-by: Alberto Nunes <bisneto.alberto@gmail.com>

* docs: fixed sintaxe, indentation and other lacking information

* docs: ✏️ WIP added email events and TODOs

* docs: ✏️ WIP progressed with some TODOs

* docs: wIP: organized commands and events

* docs: wIP:added events and emails to be sent and organized them

* Sets Yarn Version

Co-authored-by: Saulo Vallory <saulo@toki.life>
Co-authored-by: Raiane <raiane.rze@gmail.com>
Co-authored-by: Raiane Rezende <90356301+rairezende@users.noreply.github.com>
Co-authored-by: Alberto Nunes <bisneto.alberto@gmail.com>
Co-authored-by: Alberto Nunes <albnunes@users.noreply.github.com>
Co-authored-by: Gabriela Sanches <86623626+GabrielaRsanches@users.noreply.github.com>
Co-authored-by: Gabriela Sanches <gabriela@toki.life>
Co-authored-by: Raiane Rezende <raiane@toki.com>
Co-authored-by: Raquel <raquel@toki.life>

yfernandes commented:
As per @svallory instructions, this pr will be merged without approving reviews.
